src/ResourceManager/Common/Commands.ResourceManager.Common/AzureRMCmdlet.cs

```diff
@@ -204,8 +204,14 @@ protected override void InitializeQosEvent()
                 ClientRequestId = this._clientRequestId,
                 SessionId = _sessionId,
                 IsSuccess = true,
+                ParameterSetName = this.ParameterSetName
             };
 
+            if (this.MyInvocation != null && !string.IsNullOrWhiteSpace(this.MyInvocation.InvocationName))
+            {
+                _qosEvent.InvocationName = this.MyInvocation.InvocationName;
+            }
+
```

Comment:
Why not replace this whole thing (if-statement and all) with:

```
_qosEvent.InvocationName = this.MyInvocation?.InvocationName;
```

Or, can `this.MyInvocation.InvocationName` actually be whitespace?

Reply:
@twitchax I don't believe that `this.MyInvocation.InvocationName` can be whitespace, but it never hurts to make the check 😀